backend/models/group.js

```diff
@@ -246,132 +257,145 @@ groupSchema.methods.getObjectsArrayAsIfcGuids = function (data) {
 	return Promise.all(ifcGuidPromises).then(ifcObjects => {
 		return ifcObjects;
 	});
-};
+}
 
-groupSchema.statics.findIfcGroupByUID = function (dbCol, uid) {
+function ifcGuidsToUUIDs(account, model, branch, revId, ifcGuids) {
+	if(!ifcGuids || ifcGuids.length === 0) {
+		return Promise.resolve([]);
+	}
 
-	// Extract a unique list of IDs only
-	return this.findOne(dbCol, { _id: uid })
-		.then(group => {
+	const query = {"metadata.IFC GUID": { $in: ifcGuids } };
+	const project = { parents: 1, _id: 0 };
 
-			if (!group) {
-				return Promise.reject(responseCodes.GROUP_NOT_FOUND);
+	return db.getCollection(account, model + ".scene").then(dbCol => {
+		return dbCol.find(query, project).toArray().then(results => {
+			if(results.length === 0) {
+				return [];
 			}
+			return History.getHistory({ account, model }, branch, revId).then(history => {
```

Comment:
@carmenfan I agree. However, I've noticed that sometimes (e.g. issue, history) it returns MODEL_HISTORY_NOT_FOUND instead of INVALID_TAG_NAME. I guess we should pick one. I'll get rid of MODEL_HISTORY_NOT_FOUND?

Reply:
sure, agreed